src/containers/HomePage.js

```diff
@@ -4,31 +4,91 @@ import Menu from 'components/common/Menu';
 import LogoutButton from 'components/user/LogoutButton';
 import SimpleMap from 'components/common/maps/Map';
 import Welcome from 'components/user/Welcome';
+import CreateTargetForm from 'components/user/CreateTargetForm';
 import { array, func } from 'prop-types';
 
-import { loadTargets } from '../actions/targetActions';
+import { loadTargets, addTarget } from '../actions/targetActions';
 import { loadTopics } from '../actions/topicActions';
+import { FormattedMessage } from 'react-intl';
 
 class HomePage extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      targetPosition: {},
+      isCreatingNewTarget: false,
+    };
+    this.onClickMap = this.onClickMap.bind(this);
+    this.handleCreateTarget = this.handleCreateTarget.bind(this);
+  }
+
   componentDidMount() {
     this.props.loadTargets();
     this.props.loadTopics();
   }
 
+  onClickMap = ({ x, y, lat, lng, event }) => {
+    console.log(x, y, lat, lng, event);
+    const targetPosition = {
+      lat,
+      lng
+    };
+    this.setState({ targetPosition });
+    this.setState({ isCreatingNewTarget: true });
+  }
+
+  handleCreateTarget(data) {
+    const { title, radius, topicId } = data.toJS();
+    const { lat, lng } = this.state.targetPosition;
+    const targetCompleted = {
+      target: {
+        lat,
+        lng,
+        title,
+        radius,
+        topic_id: topicId[0]
```

Comment:
why are you using snake case here?

Reply:
I used a select with the prop 'multiple' then it works with an array value instead of a string. Perhaps I have to check if topicId is different of 'undefined'

Reply:
I meant, why you used `topic_id` instead of `topicId`?

Reply:
This is the body of the request that wait the api 
{
  "target": {
    "title": "new target",
    "lat": -94.5566,
    "lng": -94.5566,
    "radius": 27384.4,
    "topic_id": 2
  }
}

Reply:
Later I understand your comment, I changed it because we use decamelizeKeys.